sale_purchase_force_vendor/models/sale_order_line.py

```diff
@@ -43,7 +54,7 @@ def _prepare_procurement_values(self, group_id=False):
         if self.vendor_id:
             product = self.product_id
             suppinfo = product.with_company(self.company_id.id)._select_seller(
-                partner_id=self.vendor_id,
+                partner_id=self.vendor_id.commercial_partner_id,
```

Comment:
Don't change this, as there can be product.supplierinfo with contacts put and this will avoid its selection and will create over and over a new record:

https://github.com/OCA/purchase-workflow/blob/0c10e083ec9459b4d07c795f1cc7577d5974e8f2/sale_purchase_force_vendor/models/sale_order_line.py#L29

Maybe we should limit in the else case only to those with parent_id = False.

Reply:
Example of use case:
- There is no limit to the vendors that can be selected.
- A child partner is selected: Vendor A, Child
- When creating the `product.supplierinfo` we want "_Vendor A_" to be created.

Modifying `vendor_id_domain` to not be able to select “_children_” will be changing the behavior for some who are already using it.

Perhaps another possibility is to create a method to get the vendor and if someone needs to extend it they can do so. Any idea for the method name?

Reply:
There are reasons to have children contacts in product.supplierinfo (for example, for generating purchase orders to that children contact for communication purposes), so both the selection and the generation must be done to the children contact.

Reply:
Ok, the introduced change has been removed.